client/http.go

```diff
@@ -20,6 +22,30 @@ func request(method, url string, body io.Reader, headers *Headers) (*http.Respon
 	}
 
 	client := &http.Client{}
+
+	if localIP != nil {
+		localAddr, err := net.ResolveIPAddr("ip", *localIP)
+		if err != nil {
+			return nil, err
+		}
+
+		dialer := &net.Dialer{
```

Comment:
```
// The Client's Transport typically has internal state (cached TCP
// connections), so Clients should be reused instead of created as
// needed. Clients are safe for concurrent use by multiple goroutines.
```
From: https://golang.org/src/net/http/client.go

I'm wondering if it would be even better to allow the user to specify a custom `http.Client`. Either create one if the `client.HttpClient` is `nil`, or use the user specified http client. We can make the change backwards compatible and remove the need for passing additional args into this function.

Reply:
We can certainly do that but the issue is that all the http methods (GET, POST, request etc.) in this lib are separate functions and not on the mx client struct pointer. This means, we have to pass either the http client or the IP as an arg.

Reply:
That's ok. I think it would be a safe and simple change to make it a member of the client itself.